crates/client/src/admin_websocket.rs

```diff
@@ -61,15 +61,23 @@ impl AdminWebsocket {
     /// use std::net::Ipv4Addr;
     /// use holochain_client::AdminWebsocket;
     ///
-    /// let admin_ws = AdminWebsocket::connect((Ipv4Addr::LOCALHOST, 30_000)).await.unwrap();
+    /// let admin_ws = AdminWebsocket::connect((Ipv4Addr::LOCALHOST, 30_000), String::from("my_cli_app")).await.unwrap();
     /// # }
     /// ```
     ///
     /// As string: `"localhost:30000"`
     ///
     /// As tuple: `([127.0.0.1], 30000)`
-    pub async fn connect(socket_addr: impl ToSocketAddrs) -> ConductorApiResult<Self> {
-        Self::connect_with_config(socket_addr, Arc::new(WebsocketConfig::CLIENT_DEFAULT)).await
+    pub async fn connect(
+        socket_addr: impl ToSocketAddrs,
+        origin: Option<String>,
```

Comment:
The origin can now be specified in `AdminWebsocket::connect()`, same is in the js-client.

Reply:
I think that would mean that we need to construct a `ConnectRequest` struct and set the header on it before passing it to `connect_with_request_and_config()` wherever we want to use a custom origin. It seems a bit much boilerplate to me for something that I can see being used relatively often.

Regarding the backport to 0.5 we anyway won't be able to do that cleanly because the 0.5 branch is not using the Rust client at all to establish connection, it's still using the `CmdRunner`. I agree that we want to backport it but I think we'll need to make a dedicated PR for that with logic specific to `CmdRunner`.